recyclerview-fastscroll/src/main/java/com/simplecityapps/recyclerview_fastscroll/views/FastScrollRecyclerView.java

```diff
@@ -369,14 +374,30 @@ private void getCurScrollState(ScrollPositionState stateOut) {
         }
 
         View child = getChildAt(0);
-
         stateOut.rowIndex = getChildAdapterPosition(child);
+
+        // don't do anything if we have a header item
+        if (getLayoutManager().getChildAt(0).getHeight() < getLayoutManager().getChildAt(1).getHeight()) { // identify header
+            stateOut.rowHeight = lastRowHeight;
+            stateOut.rowIndex = lastRowIndex;
+            stateOut.rowTopOffset = lastTopOffset;
+            return;
+        }
+
         if (getLayoutManager() instanceof GridLayoutManager) {
             stateOut.rowIndex = stateOut.rowIndex / ((GridLayoutManager) getLayoutManager()).getSpanCount();
         }
         stateOut.rowTopOffset = getLayoutManager().getDecoratedTop(child);
-        stateOut.rowHeight = child.getHeight() + getLayoutManager().getTopDecorationHeight(child)
-                + getLayoutManager().getBottomDecorationHeight(child);
+        // this line here will guarantee us to have consistant height for the child
+        // we also dont need to worry about null pointer exception, since we will always have
+        // child at 1 or 0 here.
+        stateOut.rowHeight = getLayoutManager().getChildAt(0).getHeight()
+                > getLayoutManager().getChildAt(1).getHeight()
+                ? getLayoutManager().getChildAt(0).getHeight()
+                : getLayoutManager().getChildAt(1).getHeight();
+        lastRowHeight = stateOut.rowHeight;
+        lastTopOffset = stateOut.rowTopOffset;
+        lastRowIndex = stateOut.rowIndex;
```

Comment:
The same story as before, but since here lines will only be executed when we have any other items except the header item, means that the last state of recycler view will only be saved here, allows the header to load the previous state of the recycler view in the last comment. And the tertiary statement here acts as a check to identify if an item is Header or others without any hard code.